### DISCUSSION TITLE
False positive `Argument type partially is unknown` within multiple loops.

### DISCUSSION THREAD

**Discussion by andyluooo** [microsoft/pyright#4959] [Q&A] [answered]
Simplified code example: 
```python
a_value: int = any
a_list: list[int] = any

for _ in range(any):
    for _ in range(any):
        a_value = a_list[a_value]  # Argument type is partially unknown
```

It seems like a bug, which also happens in for-while loop, while-while loop, etc. 
If not, hopefully someone could help solving this. Maybe it's about possible IndexError, but not sure.

**Comment by erictraut** [comment, marked as answer]
Thanks for reporting this. It will be addressed in the next release.
